### PR TITLE
test(cat-gateway): Setup Scylla testing environment

### LINKED ISSUE #1405: 🛠️ [TASK] : Setup Scylla testing environment
Create a testing evironment to add integration test to test with EventDB.

Make sure that:
- We have a docker-compose file to run a Scylla database to test.
- We setup an Earthly target to run the integration test.

### PR DESCRIPTION
**Pull request by apskhem**
# Description

Create a testing evironment to add integration test to test with EventDB.

## Related Issue(s)

Closes #1405

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst-gateway/bin/src/db/index/tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//! Integration tests of the `IndexDB` queries
+
+#[cfg(test)]
```
Comment (stanislav-tkach) [inline catalyst-gateway/bin/src/db/index/tests/mod.rs:3]: Nitpick: I believe the attribute isn't needed here because the `tests` module already has it.

**Review by stanislav-tkach — APPROVED** [review]

File: catalyst-gateway/bin/src/db/index/tests/mod.rs (hunk 1 above)
Comment (apskhem) [reply catalyst-gateway/bin/src/db/index/tests/mod.rs]: Thank you for spotting on this. I removed this part.

File: catalyst-gateway/Earthfile [hunk 2]
```diff
@@ -139,4 +139,24 @@ test:
...
+        gpg --homedir /tmp --no-default-keyring --keyring /etc/apt/keyrings/scylladb.gpg \
+            --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys a43e06657bac99e3 && \
+        wget -O /etc/apt/sources.list.d/scylla.list http://downloads.scylladb.com/deb/debian/scylla-6.2.list && \
+        apt-get update && \
+        apt-get install -y --no-install-recommends \
+        scylla
+
+    # run the test
+    RUN --mount=$EARTHLY_RUST_CARGO_HOME_CACHE --mount=$EARTHLY_RUST_TARGET_CACHE \
+        scylla --options-file /etc/scylla/scylla.yaml --smp=2 --memory=4G --overprovisioned --developer-mode=1 & \
```
Comment (kukkok3) [inline catalyst-gateway/Earthfile:161]: I think you should copy the scylla.yaml inside the container before passing it as parameter?

File: catalyst-gateway/Earthfile [hunk 3]
```diff
@@ -139,4 +139,24 @@ test:
...
+    FROM +builder-src
+
+    # setup scylla
+    RUN apt-get update && \
+        apt-get install -y --no-install-recommends \
+        wget gnupg ca-certificates && \
+        mkdir -p /etc/apt/keyrings && \
+        gpg --homedir /tmp --no-default-keyring --keyring /etc/apt/keyrings/scylladb.gpg \
+            --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys a43e06657bac99e3 && \
+        wget -O /etc/apt/sources.list.d/scylla.list http://downloads.scylladb.com/deb/debian/scylla-6.2.list && \
```
Comment (kukkok3) [inline catalyst-gateway/Earthfile:154]: Can you make the scylla version that we pull as a earthly ARG? so we could run the same test with different scylla versions

File: catalyst-gateway/Earthfile (hunk 2 above)
Comment (apskhem) [reply catalyst-gateway/Earthfile:161]: This file is from Scylla itself after newly installing scylla. Or should I copy from the existing file instead?

File: catalyst-gateway/Earthfile (hunk 3 above)
Comment (apskhem) [reply catalyst-gateway/Earthfile:154]: I made this as separate function one, so we can select the version from this.

**Review by kukkok3 — APPROVED** [review]
LGTM

**Pull request merged**